### PR TITLE
Handle backgrounded liveblogs

### PR DESCRIPTION
**Pull request by oliverlloyd**
## What does this change?
This PR introduces logic to handle when any new blocks are revealed to the reader

## Why?
If a blog is on a page or tab that does not have focus then it will continue to poll but we don't want any updates to be inserted into the top of the blog while the page is backgrounded. Once the page becomes visible again then, at that point, the new blocks are revealed.

### CODE REVIEW

File: dotcom-rendering/src/web/components/Liveness.importable.tsx [hunk 1]
```diff
@@ -179,6 +179,42 @@ export const Liveness = ({
...
 	}
 
+	/**
+	 * This useEffect sets up a listener for when the page is backgrounded or restored. We
+	 * do this so that any new blocks that were fetched while the blog was in the
+	 * background are animated in at the point when focus is restored
+	 */
+	useEffect(() => {
+		const handleVisibilityChange = () => {
+			// The blog was either hiiden or has become visible
```
Comment (mxdvl) [inline dotcom-rendering/src/web/components/Liveness.importable.tsx:189]:
```suggestion
			// The blog was either hidden or has become visible
```

File: dotcom-rendering/src/web/components/Liveness.importable.tsx [hunk 2]
```diff
@@ -179,6 +179,42 @@ export const Liveness = ({
...
+				revealNewBlocks();
+				setNumHiddenBlocks(0);
+				setShowToast(false);
+			}
+		};
+
+		window.addEventListener(
+			'visibilitychange',
+			handleVisibilityChange,
+			false,
```
Comment (mxdvl) [inline dotcom-rendering/src/web/components/Liveness.importable.tsx:206]: Isn’t this the default value, anyways? If being explicit is what we’re after, shouldn’t we use the `options` object instead?

**Review by mxdvl — APPROVED** [review]
LGTM, with some small notes

File: dotcom-rendering/src/web/components/Liveness.importable.tsx (hunk 2 above)
Comment (oliverlloyd) [reply dotcom-rendering/src/web/components/Liveness.importable.tsx]: I think you're right, it's not required so the simplest thing is to just not pass it. Done

**Pull request merged**